apps/condo/domains/marketplace/schema/Invoice.test.js

```diff
@@ -0,0 +1,606 @@
+/**
+ * Generated by `createschema marketplace.Invoice 'number:Integer; property:Relationship:Property:PROTECT; unitType:Text; unitName:Text; accountNumber:Text; toPay:Decimal; items:Json; contact?:Relationship:Contact:SET_NULL; client?:Relationship:User:SET_NULL; clientName?:Text; clientPhone?:Text; clientEmail?:Text'`
+ */
+
+const { faker } = require('@faker-js/faker')
+const dayjs = require('dayjs')
+
+const {
+    makeLoggedInAdminClient,
+    makeClient,
+    expectValuesOfCommonFields,
+    expectToThrowAuthenticationErrorToObj,
+    expectToThrowAuthenticationErrorToObjects,
+    expectToThrowAccessDeniedErrorToObj,
+    expectToThrowGQLError,
+} = require('@open-condo/keystone/test.utils')
+
+const { CONTEXT_FINISHED_STATUS } = require('@condo/domains/acquiring/constants/context')
+const { addAcquiringIntegrationAndContext } = require('@condo/domains/acquiring/utils/testSchema')
+const {
+    addBillingIntegrationAndContext, createTestBillingProperty,
+    createRegisterBillingReceiptsPayload,
+    registerBillingReceiptsByTestClient,
+} = require('@condo/domains/billing/utils/testSchema')
+const {
+    INVOICE_STATUS_DRAFT,
+    INVOICE_STATUS_PUBLISHED,
+    INVOICE_STATUS_PAID,
+} = require('@condo/domains/marketplace/constants')
+const {
+    Invoice,
+    createTestInvoice,
+    updateTestInvoice,
+    createTestInvoiceContext,
+} = require('@condo/domains/marketplace/utils/testSchema')
+const {
+    createTestOrganization,
+    createTestOrganizationEmployeeRole,
+    createTestOrganizationEmployee,
+    makeClientWithRegisteredOrganization,
+} = require('@condo/domains/organization/utils/testSchema')
+const { FLAT_UNIT_TYPE } = require('@condo/domains/property/constants/common')
+const { makeClientWithProperty, createTestProperty } = require('@condo/domains/property/utils/testSchema')
+const { registerResidentByTestClient } = require('@condo/domains/resident/utils/testSchema')
+const {
+    makeClientWithNewRegisteredAndLoggedInUser,
+    makeClientWithSupportUser,
+    makeClientWithResidentUser,
+} = require('@condo/domains/user/utils/testSchema')
+
+let adminClient, supportClient, anonymousClient
+let dummyO10n, dummyInvoiceContext
+
+describe('Invoice', () => {
+    beforeAll(async () => {
+        adminClient = await makeLoggedInAdminClient()
+        supportClient = await makeClientWithSupportUser()
+        anonymousClient = await makeClient()
+
+        ;[dummyO10n] = await createTestOrganization(adminClient)
```

Comment:
New style 🙃

Reply:
See my answer to the above comment from @Alllex202 with some explanations.